doc/mgmt/gnmi/gnsi.md

```diff
@@ -0,0 +1,383 @@
+- /root/.ssh/authorized_users
+
+There will be only one transaction in progress, so only one copy is needed.
+###### Restore from checkpoint
+When the SSH credentials mutation session finishes without the Finalize message the state of the SSH has to be reverted to the state it was before the session started. To perform this action the Front End will send the ssh_mgmt.restore_checkpoint message to the backend that will overwrite all SSH-related files with the copy that was made when handling the ssh_mgmt.create_checkpoint message.
+Files to be restored:
+- /etc/sshd/sshd_config
+- /etc/sshd/ssh_ca_pub_key
+- /etc/sshd/ssh_host_*_cert.pub
+- /etc/sshd/ssh_host_openssh_*_key
+- /home/*/.ssh/authorized_keys
+- /home/*/.ssh/authorized_users
+- /root/.ssh/authorized_keys
+- /root/.ssh/authorized_users
+
+After the files are restored the checkpoint should be destroyed.
+
+###### Delete checkpoint
+The final step of a SSH credential mutation transaction is to delete the checkpoint that has been created at the beginning of the transaction. This operation is requested by sending the ssh_mgmt.delete_checkpoint message.
+
+###### Set credentials
+Changes of the SSH credentials are requested by sending the ssh_mgmt.set message.
+That differs from the other DBUS calls as it has a parameter - a string that contains a JSON-encoded structure that provides all details about the requested change.
+
+The structure is used to represent all types of requests and the details are shown below.
+Set authorized keys
+```json
+{ "SshAccountKeys":
+  [
+    { "account": "root",
+      "keys": [
+        { "key" : "Authorized-key #1",
+          "options": [
+            { "name": "from",
+              "value" : "*.sales.example.net,!pc.sales.example.net"
+            }
+          ]
+        },
+        { "key" : "Authorized-key #2",
+          "options": [
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+When this type of message is received the back-end will replace the content of `/home/<account>/.ssh/authorized_keys` file and restart the sshd daemon.
+Set authorized users
+```json
+{ "SshAccountUsers":
+  [
+    { "account": "root",
+      "users": [
+        { "name" : "alice",
+          "options": [
+            { "name": "from",
+              "value" : "*.sales.example.net,!pc.sales.example.net"
+            }
+          ]
+        },
+        { "name" : "bob", "options": [] }
+      ]
+    }
+  ]
+}
+```
+
+When this type of message is received the back-end will replace the content of `/home/<account>/.ssh/authorized_users` file and restart the sshd daemon.
+Set CA public key
+```json
+{ "SshCaPublicKey": [ "TEST-CERT #1", "TEST-CERT #2" ] }
+```
+
+When this type of message is received the back-end will replace the content of `/etc/sshd/ssh_ca_pub_key` file and restart the sshd daemon.
+
+### SAI API
+
+No change in SAI API.
+
+### Configuration and management
+
+#### gNMI
+
+New flags for gnmi server:
+- AuthzPolicy     bool   // Enable authz policy.
```

Comment:
To avoid confusion (thinking it is for a path of some policy file) maybe we should call it EnableAuthzPolicy. Same for PathzPolicy.

Reply:
Done